L1Trigger/L1TMuonCPPF/src/RecHitProcessor.cc

```diff
@@ -0,0 +1,462 @@
+	      l1t::CPPFDigi* MainVariables1 = new l1t::CPPFDigi(rpcId, Bx , (*cppf).int_phi, (*cppf).int_theta, isValid, (*cppf).lb, (*cppf).halfchannel, EMTFsector1, EMTFLink1, old_strip, clustersize, global_phi, global_theta);
+	      l1t::CPPFDigi* MainVariables2 = new l1t::CPPFDigi(rpcId, Bx , (*cppf).int_phi, (*cppf).int_theta, isValid, (*cppf).lb, (*cppf).halfchannel, EMTFsector2, EMTFLink2, old_strip, clustersize, global_phi, global_theta);
+	      /*
+		if(clustersize == 2 && (*cppf).int_theta == 16 && (*cppf).int_phi == 700){
+ 		std::cout << "old_rawid "<< (*cppf1).rawId << " strip " << old_strip << 
+		" firststrip "<< firststrip <<" cppf.int_phi " << (*cppf).int_phi << 
+		" cppf.int_theta " << (*cppf).int_theta << 
+		" clustersize " << clustersize << std::endl;
+                std::cout << "--------------------------------------" << std::endl;
+		}
+	      */		
+	      if ((EMTFsector1 > 0) && (EMTFsector2 == 0)){
+		cppfDigis.push_back(*MainVariables1);
+	      } 
+	      else if ((EMTFsector1 > 0) && (EMTFsector2 > 0)){
+		cppfDigis.push_back(*MainVariables1);
+		cppfDigis.push_back(*MainVariables2);
+	      }
+	      else if ((EMTFsector1 == 0) && (EMTFsector2 == 0)) {
+		continue; 
+	      } 
+	    } //Geo is true
+	    else if (!Geo){
+	      global_phi = 0.;
+	      global_theta = 0.;
+	      l1t::CPPFDigi* MainVariables1 = new l1t::CPPFDigi(rpcId, Bx , (*cppf).int_phi, (*cppf).int_theta, isValid, (*cppf).lb, (*cppf).halfchannel, EMTFsector1, EMTFLink1, old_strip, clustersize, global_phi, global_theta);
+	      l1t::CPPFDigi* MainVariables2 = new l1t::CPPFDigi(rpcId, Bx , (*cppf).int_phi, (*cppf).int_theta, isValid, (*cppf).lb, (*cppf).halfchannel, EMTFsector2, EMTFLink2, old_strip, clustersize, global_phi, global_theta);
+	      if ((EMTFsector1 > 0) && (EMTFsector2 == 0)){
+		cppfDigis.push_back(*MainVariables1);
+	      } 
+	      else if ((EMTFsector1 > 0) && (EMTFsector2 > 0)){
+		cppfDigis.push_back(*MainVariables1);
+		cppfDigis.push_back(*MainVariables2);
+	      }
+	      else if ((EMTFsector1 == 0) && (EMTFsector2 == 0)) {
+		continue;
+	      } 
+	    }
+	  } //Condition to save the CPPFDigi 
+	} //Loop over the LUTVector
+      } //Loop over the recHits
+    } // End loop: for (std::vector<const RPCRoll*>::const_iterator r = rolls.begin(); r != rolls.end(); ++r)
+  } // End loop: for (TrackingGeometry::DetContainer::const_iterator iDet = rpcGeom->dets().begin(); iDet < rpcGeom->dets().end(); iDet++)  
+  
+}
+
+
+void RecHitProcessor::process(
+			      const edm::Event& iEvent,
+			      const edm::EventSetup& iSetup,
+			      const edm::EDGetToken& recHitToken,
+			      l1t::CPPFDigiCollection& cppfDigis
+			      ) const {
+  
+  // Get the RPC Geometry
+  edm::ESHandle<RPCGeometry> rpcGeom;
+  iSetup.get<MuonGeometryRecord>().get(rpcGeom);
+  
+  // Get the RecHits from the event
+  edm::Handle<RPCRecHitCollection> recHits;
+  iEvent.getByToken(recHitToken, recHits);
+  
+  
+  // Loop over ... what, exactly?
```

Comment:
Please answer the question in the comment.

Reply:
The loop is over the detector container in the rpc geometry collection. We are interested in the RPDdetID (inside of  RPCChamber vectors), specifically, the RPCrechits. to assignment the CPPFDigis.

Reply:
Can you replace the questions with the answers please?

Reply:
done